src/notifications/v2/notifications.savedSearch.service.ts

```diff
@@ -0,0 +1,189 @@
+                await this.savedSearchNotificationService.createSavedSearchNotification(
+                    savedSearch,
+                );
+            }
+            return numberOfSearchesWithMatches;
+        }
+    };
+
+    processSavedSearchMatches = async () => {
+        console.log('Running process new saved search matches...');
+
+        const startTime = performance.now();
+
+        const yesterday = DateTime.now().minus({ days: 1 }).startOf('day');
+        const newGrants = await this.grantService.findGrantsPublishedAfterDate(
+            yesterday.toJSDate(),
+        );
+
+        console.log(
+            `Number of grants added since ${yesterday.toJSDate()}: ${
+                newGrants ? newGrants.length : 0
+            }`,
+        );
+
+        if (newGrants.length > 0) {
+            const savedSearches =
+                await this.savedSearchService.findAllConfirmedSearchesWhereDateRangeIsNullOrOverlaps(
+                    yesterday.toJSDate(),
+                );
+
+            const numberOfSearchesWithMatches =
+                await this.createSavedSearchNotifications(
+                    savedSearches,
+                    yesterday,
+                );
+
+            const endTime = performance.now();
+
+            console.log(
+                `Number of saved saved searches to process: ${
+                    savedSearches ? savedSearches.length : 0
+                }`,
+            );
+            console.log(
+                `Number of saved search notifications created: ${numberOfSearchesWithMatches}`,
+            );
+            console.log(
+                `Task took ${endTime - startTime} milliseconds to run \r\n`,
+            );
+        }
+    };
+
+    processSavedSearchMatchesNotifications = async () => {
+        console.log('Running Process Saved Search Matches Notifications...');
+
+        const startTime = performance.now();
+
+        const notifications =
+            await this.savedSearchNotificationService.getAllSavedSearchNotifications();
+
+        const batchesCount =
+            this.notificationsHelper.getNumberOfBatchesOfNotifications(
+                notifications.length,
+            );
+
+        for (let i = 0; i < batchesCount; i++) {
+            const batch = this.notificationsHelper.getBatchFromObjectArray(
+                notifications,
+                i,
+                batchesCount,
+            ) as SavedSearchNotification[];
+
+            await this.sendSavedSavedSearchNotificationEmails(batch);
+
+            console.log(`Number of emails sent: ${notifications.length}`);
+            await this.savedSearchNotificationService.deleteSentSavedSearchNotifications();
+            console.log(
+                `saved search notifications temp table has been cleared`,
+            );
+            console.log(
+                `Task took ${
+                    performance.now() - startTime
+                } milliseconds to run \r\n`,
+            );
```

Comment:
Some of these logs will be really unclear when debugging.
1. Its logging inside of a loop so will be spammy/unclear
2. Doesn't reference the scheduled job its running under
3. `saved search notifications temp table has been cleared` doesn't feel significant enough to log unless of course theres an error